service/src/main/java/bio/terra/workspace/common/exception/GcpException.java

```diff
@@ -0,0 +1,16 @@
+package bio.terra.workspace.common.exception;
+
+import bio.terra.common.exception.ErrorReportException;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import org.springframework.http.HttpStatus;
+
+/**
+ * An {@code ErrorReportException} wrapper around errors thrown by GCP. This is used to surface GCP
+ * errors directly to users.
```

Comment:
What cases should we surface GCP errors directly to users?

I would think a more specific WSM error that includes the GCP error as a cause would be a nicer error to receive and have more context.

Reply:
Looking at this again I think you're right that this is the wrong approach. If WSM unexpectedly gets a 403 on an underlying cloud resource on a request that should be correct, the WSM response really should be 500, not 403. I think there are a few spots elsewhere that we should be checking and surfacing 400s in particular, but that's really a separate issue, and I can track that separately.